app/bundles/CampaignBundle/Entity/LeadEventLogRepository.php

```diff
@@ -661,4 +662,43 @@ public function isLastFailed(int $leadId, int $eventId): bool
 
         return false;
     }
+
+    public function getEventLogStats(int $eventId): EventLogStatsDto
+    {
+        $qb = $this->getReplicaConnection()->createQueryBuilder();
+        $qb->select([
+            'COUNT(log.id) as total_logs',
+            'SUM(log.is_scheduled) as pending_executions',
+            'SUM(CASE WHEN log.non_action_path_taken = 1 AND log.is_scheduled = 0 THEN 1 ELSE 0 END) as negative_path_count',
+            'SUM(CASE WHEN log.non_action_path_taken = 0 AND log.is_scheduled = 0 THEN 1 ELSE 0 END) as positive_path_count',
+            'MIN(log.date_triggered) as first_execution_date',
+            'MAX(log.date_triggered) as last_execution_date',
```

Comment:
Looking at this I'm thinking how to implement rotations to this. The execution can be 100 but it could be just 10 contacts rotating through the campaign. It may be a good information for the marketer also for unique executions. And perhaps also max rotations? WDYT?

Reply:
Good point, I've added unique runs and max rotation stats. It will be visible only if the campaign allows restart.